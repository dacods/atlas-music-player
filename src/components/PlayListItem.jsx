export default function PlayListItem({ title, artist, length, selected }) {
  return (
    <div
  className={`flex w-full items-center justify-between rounded-lg px-0 py-1 ${
    selected ? "bg-beige dark:bg-dark-beige text-black" : ""
  }`}
>
  <div className="flex flex-col">
    <p className="font-semibold">{title}</p>
    <p className="text-sm text-charcoal-gray">{artist}</p>
  </div>
  <div className="text-sm text-charcoal-gray whitespace-nowrap pl-4 shrink-0">
    {length}
  </div>
</div>
  );
}