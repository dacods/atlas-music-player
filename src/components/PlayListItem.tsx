type PlayListItemProps = {
  title: string;
  artist: string;
  length: string;
  selected?: boolean;
  onClick?: () => void;
}

export default function PlayListItem({ title, artist, length, selected=false, onClick }: PlayListItemProps) {
  return (
    <div
  className={`flex w-full items-center justify-between rounded-lg px-0 py-1 ${
    selected ? "bg-beige dark:bg-dark-beige text-black" : ""
  }`} onClick={onClick}
>
  <div className="flex flex-col">
    <p className="font-semibold">{title}</p>
    <p className="text-sm text-charcoal-gray">{artist}</p>
  </div>
  <div className="text-sm text-charcoal-gray whitespace-nowrap pl-4 shrink-0">
    {length}
  </div>
</div>
  );
}