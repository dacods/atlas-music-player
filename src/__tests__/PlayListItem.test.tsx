import { test, expect } from 'vitest';
import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

test('renders PlayListItem', () => {
    const { container } = render(<PlayListItem
    title='Song 1'
    artist='Artist 1'
    length='3:41'
    selected={false}
    onClick={() => {}}
    ></PlayListItem>)
    expect(container).toMatchSnapshot();
})