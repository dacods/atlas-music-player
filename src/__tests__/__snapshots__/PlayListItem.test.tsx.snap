// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PlayListItem 1`] = `
<div>
  <div
    class="flex w-full items-center justify-between rounded-lg px-0 py-1 "
  >
    <div
      class="flex flex-col"
    >
      <p
        class="font-semibold"
      >
        Song 1
      </p>
      <p
        class="text-sm text-charcoal-gray"
      >
        Artist 1
      </p>
    </div>
    <div
      class="text-sm text-charcoal-gray whitespace-nowrap pl-4 shrink-0"
    >
      3:41
    </div>
  </div>
</div>
`;
